### PR TITLE
 For #10683 - Show long tab titles on two lines 

### LINKED ISSUE #10683: [Tab Tray] Show longer titles (on 2 rows)..., to match current Fenix Home Screen
## Steps to reproduce
Activate new Tab Tray secret setting
Open some sites to feed the Tab Tray :grinning:
Move to Tab Tray

### Expected behavior
Titles displayed for sites should be long enough to be meaningful (note that I don't rely on Site Previews where the expected information could not present/readable).
IMHO, the current Fenix Home Page as shown below shows a good example of what the information density should be:

* number of rows (2 rows for the title), space between rows
* font size & color (contrast) differences between titles and URLs.

image

### Actual behavior
Titles displayed for sites are truncated on 1 row each, therefore they are not meaningful at all, as you can see below (same sites as above, in reverse order)

image

### Device information
* Android device: 9
* Fenix version: build #2015740403

Thank you and pardon my French :wink:

### PR DESCRIPTION
**Pull request by amkcpu**
Closes #10683. 
This is achieved by doing three things:
1. Constrain the tab title and tab URL `TextView`s to the parent's top and bottom, centering them vertically in the `tab_item` `ContraintLayout`.
2. Add TextView autosizing to the tab title `TextView` to account for both situations, when the title has only one line and its maximum of two lines (this maximum is set with `android:maxLines`). Autosizing is supported all the way down to Android 4.0 (API Level 14), see details [here](https://developer.android.com/guide/topics/ui/look-and-feel/autosizing-textview).
3. Used a "packed" vertical chain to make sure that the tab title and tab URL stick together.

**Screenshot:**

**Note**
I did not touch the thumbnail but I believe it would be good if it too was simply constrained vertically to the parent's top and bottom to make sure is centers exactly in the middle (vertically) of the `tab_item` `ContraintLayout`. Though that should probably be left for another issue/PR.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by mcarare** [comment]
This still requires UX to sign off on the change. Also, I do not see any mention of auto-sizing the text, I doubt UX will approve that.

File: app/src/main/res/layout/tab_tray_item.xml [hunk 1]
```diff
@@ -57,15 +57,18 @@
         android:layout_height="wrap_content"
```
Comment (mcarare) [inline app/src/main/res/layout/tab_tray_item.xml:57]: This is not advisable to be used with auto sizing.

File: app/src/main/res/layout/tab_tray_item.xml [hunk 2]
```diff
@@ -57,15 +57,18 @@
...
         android:layout_alignParentTop="true"
         android:ellipsize="end"
-        android:lines="1"
         android:paddingStart="16dp"
-        android:paddingTop="22dp"
         android:textColor="@color/tab_tray_item_text_normal_theme"
         android:textSize="16sp"
```
Comment (mcarare) [inline app/src/main/res/layout/tab_tray_item.xml:62]: This is redundant with auto sizing.

Comment (mcarare) [inline app/src/main/res/layout/tab_tray_item.xml:62]: This is redundant with auto sizing.

File: app/src/main/res/layout/tab_tray_item.xml (hunk 1 above)
Comment (amkcpu) [reply app/src/main/res/layout/tab_tray_item.xml:57]: Do you have a suggestion what should be used as `layout_height` value instead?

Comment (mcarare) [reply app/src/main/res/layout/tab_tray_item.xml:57]: If UX approves using auto sizing, they should also specify a fixed dp height for this in the specs.

Comment (amkcpu) [reply app/src/main/res/layout/tab_tray_item.xml:57]: I will wait for UX to weight in then

File: app/src/main/res/layout/tab_tray_item.xml (hunk 2 above)
Comment (amkcpu) [reply app/src/main/res/layout/tab_tray_item.xml]: I added a commit for this

**Comment by liuche** [comment]
Could UX chime in here what the height of the spec should be? so that we don't have to use auto-sizing

**Comment by klint** [comment]
Any update on this PR? 😊 Thanks

**Comment by liuche** [comment]
@topotropic do you have any feedback here on what the max height of a 2-line title should be? Or a 1-line title.

**Comment by amkcpu** [comment]
Closing this PR in favor of #15886.

**Pull request closed without merging**